samigo/samigo-qti/src/java/org/sakaiproject/tool/assessment/qti/helper/AuthoringHelper.java

```diff
@@ -679,43 +697,205 @@ public AssessmentFacade createImportedAssessment(Document document, String unzip
+              item2.setSection(null);
+              item2.setSequence(null);
+              ItemFacade itemFacade = itemService.saveItem(new ItemFacade(item2));
+              //item.setItemId(item.getItemId());
+              //item.setItemIdString(item.getItemId().toString());
+              
+              QuestionPoolItemData questionPoolItem = new QuestionPoolItemData();
+              questionPoolItem.setQuestionPoolId(pool.getQuestionPoolId());
+              questionPoolItem.setItemId(itemFacade.getItemId());        
+              pool.addQuestionPoolItem((QuestionPoolItemIfc) questionPoolItem);
+            }
+            questionPoolService.savePool(pool);
           }
-          // make sure required fields are set
-          item.setCreatedBy(me);
-          item.setCreatedDate(assessment.getCreatedDate());
-          item.setLastModifiedBy(me);
-          item.setLastModifiedDate(assessment.getCreatedDate());
-          item.setStatus(ItemDataIfc.ACTIVE_STATUS);
-          // assign the next sequence number
-          item.setSequence(  Integer.valueOf(itm + 1));
-          // add item to section
-          item.setSection(section); // one to many
-          // update metadata with PARTID
-          item.addItemMetaData(ItemMetaData.PARTID, section.getSectionId().toString());
-           
-          // Item Attachment
-          exHelper.makeItemAttachmentSet(item);
           
-          item = itemService.saveItem(item);
-          section.addItem(item);
-          EventTrackingService.post(EventTrackingService.newEvent(SamigoConstants.EVENT_ASSESSMENT_SAVEITEM, "/sam/" + AgentFacade.getCurrentSiteId() + "/saved itemId=" + item.getItemId().toString(), true));
-        } // ... end for each item
-        
+        } // This assessment is not based on questionpool
+        else {
+          
+          List itemList = exHelper.getItemXmlList(sectionXml);
+          for (int itm = 0; itm < itemList.size(); itm++) // for each item
+          {
+            log.debug("items=" + itemList.size());
+            Item itemXml = (Item) itemList.get(itm);
+            
+            ItemFacade item = new ItemFacade();
+          try {
+            exHelper.updateItem(item, itemXml, isRespondus);
+          }
+          catch (RespondusMatchingException rme) {
+            if (failedMatchingQuestions != null) {
+              failedMatchingQuestions.add(itm + 1);
+            }
+          }
+            
+            // make sure required fields are set
+            item.setCreatedBy(me);
+            item.setCreatedDate(assessment.getCreatedDate());
+            item.setLastModifiedBy(me);
+            item.setLastModifiedDate(assessment.getCreatedDate());
+            item.setStatus(ItemDataIfc.ACTIVE_STATUS);
+            // assign the next sequence number
+            item.setSequence(  Integer.valueOf(itm + 1));
+            // add item to section
+            item.setSection(section); // one to many
```

Comment:
I think some of this comments could be removed

Reply:
Removed